### ISSUE TITLE
[no-negated-async] should double-bang be triggering this rule?

### ISSUE THREAD

**Issue by mayhew3** [angular-eslint/angular-eslint#404]
**Description and reproduction of the issue**

The double bang (!!) seems to be a fairly standard way of testing for a truthy value. From what I've read, the no-negated-async rule exists because an async returns null while it is waiting, and so calling ! on it will over-fire from the template. However, doesn't that mean the double-bang will actually UNDER-fire, and so is actually safe?

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint       `        | `4.16.1` |
| `TypeScript`                       | `4.1.5` |
| `ESLint`                           | `7.6.0` |
| `node`                             | `14.15.0` |

```sh
# Please run `npx ng version` in your project and paste the full output here:

Angular CLI: 11.2.5
Node: 14.15.0
OS: win32 x64

Angular: 11.2.6
... animations, cdk, common, compiler, compiler-cli, core, forms
... language-service, localize, material, platform-browser
... platform-browser-dynamic, router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1102.5
@angular-devkit/build-angular   0.1102.5
@angular-devkit/core            11.2.5
@angular-devkit/schematics      11.2.5
@angular/cli                    11.2.5
@schematics/angular             11.2.5
@schematics/update              0.1102.5
rxjs                            6.6.6
typescript                      4.1.5

```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.
